components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/OAuthScopeDAOImpl.java

```diff
@@ -444,13 +456,15 @@ public void deleteScopeByName(String name, int tenantID) throws IdentityOAuth2Sc
     public void updateScopeByName(Scope updatedScope, int tenantID) throws IdentityOAuth2ScopeServerException {
 
         if (log.isDebugEnabled()) {
-            log.debug("Updae scope by name for scope name:" + updatedScope.getName());
+            log.debug("Update scope by name for scope name:" + updatedScope.getName());
         }
 
         try (Connection conn = IdentityDatabaseUtil.getDBConnection()) {
             try {
-                deleteScope(updatedScope.getName(), tenantID, conn);
-                addScope(updatedScope, conn, tenantID);
+                int scopeId = getScopeId(updatedScope.getName(), tenantID, conn);
+                updateScopeDetails(updatedScope, conn, scopeId);
+                deleteBindings(updatedScope.getName(), tenantID, conn);
```

Comment:
Better to use scopeID to delete the binding.

Reply:
fixed